delta-api/src/main/java/io/cdap/delta/api/assessment/TableNotFoundException.java

```diff
@@ -25,7 +25,19 @@ public TableNotFoundException(String database, String table, String errorMessage
     super(String.format("Table '%s' in database '%s' was not found. %s", table, database, errorMessage));
   }
 
+  public TableNotFoundException(String database, String schema, String table, String errorMessage) {
+    super(String.format("Table '%s' in database '%s' and schema '%s' was not found. %s", table, database, schema,
```

Comment:
We can form a message in a variable here depending on the schema is null or not and then use that variable to call super.

Reply:
My major concern is that assembling the message based on the schema will make the code clumsy and less readable. because the schema information appears in the middle of the message.

Reply:
also call to super must be the first statement in the constructor.